src/main/java/org/jboss/threads/EnhancedQueueExecutor.java

```diff
@@ -1749,6 +1778,10 @@ private int tryExecute(final Runnable runnable) {
                     // post-actions (fail):
                     //   retry outer with new tail(snapshot)
                     if (consumerNode.compareAndSetTask(WAITING, runnable)) {
+                        // GC Nepotism:
+                        // We can save consumerNode::next from being dragged into
+                        // old generation, if possible
+                        consumerNode.compareAndSetNext(tailNextNext, null);
```

Comment:
I've used a compareAndSetNext because I'm not sure that this case is single-writer, maybe yes.

Reply:
Maybe we should do a weakCompareAndSet here to avoid the heavier memory barrier?

Reply:
IMO is safe wekCompareAndSet, but probably will make any difference only for ARM/PowerPC servers: AFAIK newish x86s won't have any specific code path implemented on the OpenJDK for this (but I could be wrong)

Reply:
OK.  Well we do care about ARM and AArch64 so I think that makes it valid.